### PR TITLE
Remove static variables for PHP 8.1

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- make PHP 8.1 compatible
  - see https://github.com/codeigniter4/CodeIgniter4/pull/4883#issuecomment-955622746

See "Usage of static Variables in Inherited Methods"
https://www.php.net/manual/en/migration81.incompatible.php

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/BaseBuilder.php [hunk 1]
```diff
@@ -2515,13 +2529,11 @@ protected function isLiteral(string $str): bool
             return true;
         }
 
-        static $_str;
-
-        if (empty($_str)) {
-            $_str = ($this->db->escapeChar !== '"') ? ['"', "'"] : ["'"];
+        if (empty($this->isLiteralStr)) {
+            $this->isLiteralStr = ($this->db->escapeChar !== '"') ? ['"', "'"] : ["'"];
```
Comment (paulbalandan) [inline system/Database/BaseBuilder.php:2533]:
```suggestion
            $this->isLiteralStr = $this->db->escapeChar !== '"' ? ['"', "'"] : ["'"];
```

File: system/Database/BaseBuilder.php [hunk 2]
```diff
@@ -2600,7 +2612,10 @@ protected function resetWrite()
      */
     protected function hasOperator(string $str): bool
     {
-        return (bool) preg_match('/(<|>|!|=|\sIS NULL|\sIS NOT NULL|\sEXISTS|\sBETWEEN|\sLIKE|\sIN\s*\(|\s)/i', trim($str));
+        return (bool) preg_match(
+            '/(<|>|!|=|\sIS NULL|\sIS NOT NULL|\sEXISTS|\sBETWEEN|\sLIKE|\sIN\s*\(|\s)/i',
+            trim($str)
+        );
```
Comment (paulbalandan) [inline system/Database/BaseBuilder.php:2618]:
```suggestion
        return preg_match(
            '/(<|>|!|=|\sIS NULL|\sIS NOT NULL|\sEXISTS|\sBETWEEN|\sLIKE|\sIN\s*\(|\s)/i',
            trim($str)
        ) === 1;
```

File: system/Database/BaseBuilder.php [hunk 3]
```diff
@@ -2610,11 +2625,11 @@ protected function hasOperator(string $str): bool
      */
     protected function getOperator(string $str, bool $list = false)
     {
-        static $_operators;
-
-        if (empty($_operators)) {
-            $_les       = ($this->db->likeEscapeStr !== '') ? '\s+' . preg_quote(trim(sprintf($this->db->likeEscapeStr, $this->db->likeEscapeChar)), '/') : '';
-            $_operators = [
+        if (empty($this->pregOperators)) {
+            $_les = ($this->db->likeEscapeStr !== '')
```
Comment (paulbalandan) [inline system/Database/BaseBuilder.php:2629]:
```suggestion
            $_les = $this->db->likeEscapeStr !== ''
```

File: system/Database/BaseBuilder.php [hunk 4]
```diff
@@ -2515,13 +2529,11 @@ protected function isLiteral(string $str): bool
             return true;
         }
 
-        static $_str;
-
-        if (empty($_str)) {
-            $_str = ($this->db->escapeChar !== '"') ? ['"', "'"] : ["'"];
+        if (empty($this->isLiteralStr)) {
```
Comment (paulbalandan) [inline system/Database/BaseBuilder.php:2532]:
```suggestion
        if ($this->isLiteralStr === []) {
```

File: system/Database/BaseBuilder.php [hunk 5]
```diff
@@ -2610,11 +2625,11 @@ protected function hasOperator(string $str): bool
      */
     protected function getOperator(string $str, bool $list = false)
     {
-        static $_operators;
-
-        if (empty($_operators)) {
-            $_les       = ($this->db->likeEscapeStr !== '') ? '\s+' . preg_quote(trim(sprintf($this->db->likeEscapeStr, $this->db->likeEscapeChar)), '/') : '';
-            $_operators = [
+        if (empty($this->pregOperators)) {
```
Comment (paulbalandan) [inline system/Database/BaseBuilder.php:2628]:
```suggestion
        if ($this->pregOperators === []) {
```

File: system/Database/BaseConnection.php [hunk 6]
```diff
@@ -1119,29 +1126,35 @@ public function escapeIdentifiers($item)
             return $item;
         }
 
-        static $pregEc = [];
-
-        if (empty($pregEc)) {
+        if (empty($this->pregEscapeChar)) {
```
Comment (paulbalandan) [inline system/Database/BaseConnection.php:1129]:
```suggestion
        if ($this->pregEscapeChar === []) {
```

**Review by paulbalandan — APPROVED** [review]

**Comment by MGatner** [comment]
Glad to see these go. Inline static variables in classes weird me out.

**Pull request merged**